src/mimir_config.py

```diff
@@ -84,11 +87,11 @@ class Alertmanager(BaseModel):
 
 
 class _S3StorageBackend(BaseModel):
```

Comment:
If we're importing this from another module, it should not start with an underscore.

[also, a nitpick] I'd rename it to `S3BucketConfig` or `S3StorageBackendData`, because it's not really a 'backend': it's the config/data you need to connect to one